src/libs/actions/Link.js

```diff
@@ -49,8 +50,13 @@ function openOldDotLink(url) {
 
         const params = _.compact([authTokenParam, emailParam]).join('&');
 
-        // If the URL contains # or ?, we can assume they don't need to have the `?` token to start listing url parameters.
-        return `${CONFIG.EXPENSIFY.EXPENSIFY_URL}${url}${hasHashParams || hasURLParams ? '&' : '?'}${params}`;
+        return Environment.getOldDotEnvironmentURL()
+            .then((environmentURL) => {
+                const oldDotDomain = Url.addTrailingForwardSlash(environmentURL);
+
+                // If the URL contains # or ?, we can assume they don't need to have the `?` token to start listing url parameters.
+                return `${oldDotDomain}${url}${hasHashParams || hasURLParams ? '&' : '?'}${params}`;
+            });
     }
 
     if (isNetworkOffline) {
```

Comment:
@MonilBhavsar A bit late, but don't we need to make changes here? `Linking.openURL` expects a string not a promise.
```js
        buildOldDotURL().then((oldDotUrl) => {
            Linking.openURL(oldDotUrl);
        });

```

Reply:
We missed it. It was fixed here https://github.com/Expensify/App/pull/15923

Reply:
Oh too late then. Sorry about that